### ISSUE TITLE
@angular/language-service slows down VSCode autocomplete on every keystroke

### ISSUE THREAD

**Issue by gund** [angular/angular#27904]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/language-service

### Is this a regression?

 Yes, the previous versions did not cause the lag but I cannot install specific version in vscode because it's always using latest version even if plugin version is not.

### Description

 Every time I modify TS file or try to navigate to a symbol that is imported from another TS file `language-service` is causing autocompletion to lag for 3-4 seconds.

Tested in mid size project (~500 components, ~100 modules).

## 🔬 Minimal Reproduction

- Open any angular project in VSCode with Angular Language Service extension installed and enabled
- Open output of Angular Language Service extension (Cmd+Shift+U -> Select Angular Language Service in dropdown in top right corner)
- Open any TS file from `src/app/` folder
- Hover over any token, you will see that is takes about 1ms to complete a task
- Now type something in and wait a bit, you will see `getDiagnostics: 4547ms` message that will appear with delay, and during this period autocomplete in TS files will not load.

## 🔥 Exception or Error
Here is the log of Angular Language Service:
<pre><code>
Search path: /Users/alex/Proj/mcmakler-admin-v2/src/app/shared/directives/assign-as
Config file name: /Users/alex/Proj/mcmakler-admin-v2/tsconfig.json
Angular Language Service: 7.1.4
TypeScript: 3.1.6
getTemplateReferences: 6125ms
Add recursive watcher for: /Users/alex/Proj/mcmakler-admin-v2
Opened configuration file /Users/alex/Proj/mcmakler-admin-v2/tsconfig.json
getDiagnostics: 8ms
getHoverAt: 1ms
getDefinitionAt: 1ms
Search path: /Users/alex/Proj/mcmakler-admin-v2/node_modules/@angular/core/src/metadata
Config file name: /Users/alex/Proj/mcmakler-admin-v2/tsconfig.json
getTemplateReferences: 0ms
getHoverAt: 0ms
getDiagnostics: 0ms
getCompletions: 638ms
getDiagnostics: 4347ms // <-- Slowdown of 4s after keystroke
updating project structure from ...
updated: ng - 0ms
getDiagnostics: 842ms
updating project structure from ...
updated: ng - 0ms
getCompletions: 3902ms // <-- Slowdown of 4s after keystroke
getDiagnostics: 251ms
updating project structure from ...
updated: ng - 0ms
getDiagnostics: 822ms
updating project structure from ...
updated: ng - 0ms
getDiagnostics: 4380ms // <-- Slowdown of 4s after navigating to external symbol (TemplateRef)
updating project structure from ...
updated: ng - 0ms
getHoverAt: 1ms
getHoverAt: 0ms
</code></pre>

## 🌍  Your Environment

**Angular Version:**
<pre><code>
Angular CLI: 7.1.2
Node: 10.14.2
OS: darwin x64
Angular: 7.1.3
... animations, common, compiler, compiler-cli, core, forms
... http, platform-browser, platform-browser-dynamic, router
... upgrade

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.11.2
@angular-devkit/build-angular     0.11.2
@angular-devkit/build-optimizer   0.11.2
@angular-devkit/build-webpack     0.11.2
@angular-devkit/core              7.1.2
@angular-devkit/schematics        7.1.2
@angular/cdk                      7.1.1
@angular/cli                      7.1.2
@angular/flex-layout              7.0.0-beta.19
@angular/language-service         7.1.4
@angular/material                 7.1.1
@ngtools/webpack                  7.1.2
@schematics/angular               7.1.2
@schematics/update                0.11.2
rxjs                              6.3.3
typescript                        3.1.6
webpack                           4.23.1
</code></pre>

**Anything else relevant?**

OS: MacOS Mojave
IDE: VSCode

**Comment by gund** [comment]
Saving a file also causing same lags.

In the logs I get:
```
Search path: /Users/alex/Proj/mcmakler-admin-v2/src/app/shared/directives/assign-as
Config file name: /Users/alex/Proj/mcmakler-admin-v2/tsconfig.json
getTemplateReferences: 0ms
getCompletions: 438ms
getDiagnostics: 4085ms // <-- Slowdown of 4s after saving a file and triggering autocomplete
updating project structure from ...
updated: ng - 0ms
```

**Comment by kyliau** [comment]
Thanks for the report, I'll investigate the issue this week with high priority.
This only happens to a subset of our users, which I believe is caused by Typescript compiler version mismatch. I'll update ticket with more info later this week.

**Comment by j34nt** [comment]
any updates???

**Comment by kyliau** [comment]
I've published a beta version, and from the testing that I've done so far, it works well with Angular v7.
Could you please try it out and let me know if it fixes the issue for you?
https://github.com/angular/vscode-ng-language-service/releases/tag/v0.800.0-rc.0

**Comment by jamiemac87** [comment]
@kyliau the beta version you published works for me and I'm not seeing anymore lag

**Comment by kyliau** [comment]
@jamiemac87 Thank you so much for the confirmation! 
If you run into any issues at all, please open an issue and ping me.
I'll push the new version to the Marketplace once Anguar v8 stable is released this week.

**Comment by StefVanremoortele** [comment]
@kyliau i've tested your beta version too, it works and i'm truly thankfull.
For me it was getCompletions that was causing the lag.
thanks!

**Comment by kyliau** [comment]
Related on the extension repo: https://github.com/angular/vscode-ng-language-service/issues/329

**Comment by kyliau** [comment]
Released v0.800.0 that fixed performance regression.
Multiple beta users confirmed that language service works well with Angular v7 and v8.
Please open another issue if this is not the case.
https://github.com/angular/vscode-ng-language-service/releases/tag/v0.800.0

Related issues:
- https://github.com/angular/vscode-ng-language-service/issues/329
- https://github.com/angular/angular/issues/27904

**Comment by smasala** [comment]
@kyliau I'm getting slow downs for `getTemplateReferences: 8484ms`

```
Angular Language Service: 8.0.0
TypeScript: 3.4.5
```
